.clang-format

```diff
@@ -1,140 +1,10 @@
+# Documentation about the style options and their meaning
```

Comment:
Assume this is somehow meant for this PR?

Reply:
Also, is there a current lint checker in CI? Might be useful to add a pre-commit action to GitHub. Wouldn't take more than an hour or two...